packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -99,48 +84,67 @@ class DevFSEntry {
     }
   }
 
-  File _getFile() {
-    if (_linkTarget != null) {
-      return _linkTarget;
-    }
-    if (file is Link) {
-      // The link target.
-      return fs.file(file.resolveSymbolicLinksSync());
-    }
-    return file;
+  @override
+  bool get isModified {
+    FileStat _oldFileStat = _fileStat;
+    _stat();
+    return _oldFileStat == null || _fileStat.modified.isAfter(_oldFileStat.modified);
   }
 
-  Future<List<int>> contentsAsBytes() async {
-    if (_isSourceEntry)
-      return bundleEntry.contentsAsBytes();
-    final File file = _getFile();
-    return file.readAsBytes();
+  @override
+  int get size {
+    if (_fileStat == null)
+      _stat();
+    return _fileStat.size;
   }
 
-  Stream<List<int>> contentsAsStream() {
-    if (_isSourceEntry) {
-      return new Stream<List<int>>.fromIterable(
-          <List<int>>[bundleEntry.contentsAsBytes()]);
-    }
-    final File file = _getFile();
-    return file.openRead();
-  }
+  @override
+  Future<List<int>> contentsAsBytes() => _getFile().readAsBytes();
 
-  Stream<List<int>> contentsAsCompressedStream() {
-    return contentsAsStream().transform(GZIP.encoder);
+  @override
+  Stream<List<int>> contentsAsStream() => _getFile().openRead();
+}
+
+/// Byte content to be copied to the device.
+class DevFSByteContent extends DevFSContent {
+  DevFSByteContent(this.bytes);
+
+  final List<int> bytes;
```

Comment:
does it make sense to let the DevFSByteContent mutable? And whenever bytes it set modified is set to true again?